src/SFA.DAS.ProviderCommitments/SFA.DAS.ProviderCommitments.Web/Controllers/ApprenticeController.cs

```diff
@@ -23,45 +25,41 @@ public ApprenticeController(IModelMapper modelMapper)
             _modelMapper = modelMapper;
         }
 
-        [Route("", Name = RouteNames.ApprenticesIndex)]
-        [DasAuthorize(ProviderFeature.ManageApprenticesV2)]
-        public async Task<IActionResult> Index(IndexRequest request)
-        {
-            var viewModel = await _modelMapper.Map<IndexViewModel>(request);
-            viewModel.SortedByHeader();
-
-            return View(viewModel);
-        }
-
-        [Route("{apprenticeshipHashedId}", Name = RouteNames.ApprenticeDetail)]
-        [DasAuthorize(CommitmentOperation.AccessApprenticeship, ProviderFeature.ApprenticeDetailsV2)]
-        public async Task<IActionResult> Details(DetailsRequest request)
+        [HttpGet]
+        [Route("{apprenticeshipHashedId}/change-employer/price", Name = RouteNames.ApprenticeChangePrice)]
+        [DasAuthorize(CommitmentOperation.AccessApprenticeship, ProviderFeature.ChangeOfEmployer)]
+        public Task<IActionResult> ChangePrice(PriceRequest request)
         {
-            var viewModel = await _modelMapper.Map<DetailsViewModel>(request);
-            return View(viewModel);
+            throw new NotImplementedException();
         }
 
         [HttpGet]
-        [Route("{apprenticeshipHashedId}/change-employer", Name = RouteNames.ApprenticeInform)]
+        [Route("{apprenticeshipHashedId}/change-employer/dates", Name = RouteNames.ApprenticeDates)]
         [DasAuthorize(CommitmentOperation.AccessApprenticeship, ProviderFeature.ChangeOfEmployer)]
-        public async Task<IActionResult> Inform(InformRequest request)
+        public async Task<IActionResult> Dates(DatesRequest request)
         {
-            var viewModel = await _modelMapper.Map<InformViewModel>(request);
+            var viewModel = await _modelMapper.Map<DatesViewModel>(request);
 
             return View(viewModel);
         }
 
-        [HttpGet]
-        [Route("{apprenticeshipHashedId}/change-employer/select-employer", Name = RouteNames.ApprenticeSelectEmployer)]
+        [HttpPost]
+        [Route("{apprenticeshipHashedId}/change-employer/dates", Name = RouteNames.ApprenticeDates)]
         [DasAuthorize(CommitmentOperation.AccessApprenticeship, ProviderFeature.ChangeOfEmployer)]
-        public async Task<IActionResult> SelectEmployer(SelectEmployerRequest request)
+        public async Task<IActionResult> Dates(DatesViewModel viewModel)
         {
-            var viewModel = await _modelMapper.Map<SelectEmployerViewModel>(request);
-
-            return View(viewModel);
+            try
+            {
+                var request = await _modelMapper.Map<PriceRequest>(viewModel);
+                return RedirectToAction(nameof(ChangePrice), request);
+            }
+            catch (ValidationException)
```

Comment:
We shouldn't have to do this kind of thing, as the PRG implementation should handle it for us.

Reply:
removed